src/ml/neural_net/mps_layer_instance_norm_data_loader.mm

```diff
@@ -0,0 +1,202 @@
+                                  options:MTLResourceStorageModeManaged];
+
+
+    _gammaMomentumBuffer = [dev newBufferWithBytes:zeros_ptr
+                                            length:sizeof(float) * _numberOfFeatureChannels
+                                           options:MTLResourceStorageModeManaged];
+
+    _gammaVelocityBuffer = [dev newBufferWithBytes:zeros_ptr
+                                            length:sizeof(float) * _numberOfFeatureChannels
+                                           options:MTLResourceStorageModeManaged];
+
+    _betaMomentumBuffer = [dev newBufferWithBytes:zeros_ptr
+                                           length:sizeof(float) * _numberOfFeatureChannels
+                                          options:MTLResourceStorageModeManaged];
+
+    _betaVelocityBuffer = [dev newBufferWithBytes:zeros_ptr
+                                           length:sizeof(float) * _numberOfFeatureChannels
+                                          options:MTLResourceStorageModeManaged];
+
+    _movingMeanBuffer = [dev newBufferWithBytes:zeros_ptr
+                                         length:sizeof(float) * _numberOfFeatureChannels
+                                        options:MTLResourceStorageModeManaged];
+
+    _movingVarianceBuffer = [dev newBufferWithBytes:ones_ptr
+                                             length:sizeof(float) * _numberOfFeatureChannels
+                                            options:MTLResourceStorageModeManaged];
+    
+    _adamGamma = [[MPSNNOptimizerAdam alloc] initWithDevice:dev
+                                              learningRate:0.001f];
+        
+    _adamBeta = [[MPSNNOptimizerAdam alloc] initWithDevice:dev
+                                             learningRate:0.001f];
+      
+    _gammaVector = [[MPSVector alloc] initWithBuffer:_gammaBuffer
+                                          descriptor:_vDesc];
+
+    _gammaMomentumVector = [[MPSVector alloc] initWithBuffer:_gammaMomentumBuffer
+                                                  descriptor:_vDesc];
+
+    _gammaVelocityVector = [[MPSVector alloc] initWithBuffer:_gammaVelocityBuffer
+                                                  descriptor:_vDesc];
+
+    _betaVector = [[MPSVector alloc] initWithBuffer:_betaBuffer
+                                         descriptor:_vDesc];
+
+    _betaMomentumVector = [[MPSVector alloc] initWithBuffer:_betaMomentumBuffer
+                                                 descriptor:_vDesc];
+
+    _betaVelocityVector = [[MPSVector alloc] initWithBuffer:_betaVelocityBuffer
+                                                 descriptor:_vDesc];
+
+
+    _state = [[MPSCNNNormalizationGammaAndBetaState alloc] initWithGamma:_gammaBuffer
+                                                                    beta:_betaBuffer];
+  }
+  return self;
+}
+
+- (void) updateNumberOfStyles:(NSUInteger)styles {
```

Comment:
Hmm, looks like this just be exposed as a public property. So for this one, in the header:
```
@property (nonatomic) NSUInteger stylesCount;
```

Then in the .m file
```
- (void)setStylesCount:(NSUInteger)count {
  _stylesCount = count;
}
```

Reply:
changed.